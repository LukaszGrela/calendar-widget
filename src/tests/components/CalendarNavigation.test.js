import React from 'react';
import { shallow } from 'enzyme'
import CalendarNavigation from '../../components/CalendarNavigation';

let wrapper, navigateUp, navigateDown;

beforeEach(() => {
    navigateUp = jest.fn();
    navigateDown = jest.fn();
    wrapper = shallow(<CalendarNavigation
        navigateDown={navigateDown}
        navigateUp={navigateUp}
    />);
});

test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly with given className', () => {
    wrapper.setProps({
        className: 'my-class'
    })
    expect(wrapper).toMatchSnapshot();
});


test('should call navigateUp', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(navigateUp).toHaveBeenCalled();
});

test('should call navigateDown', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(navigateDown).toHaveBeenCalled();
});