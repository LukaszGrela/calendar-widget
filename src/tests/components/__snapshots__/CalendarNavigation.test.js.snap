// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="month-navigation"
>
  <button
    onClick={[Function]}
  >
    <IconUp />
  </button>
  <button
    onClick={[Function]}
  >
    <IconDown />
  </button>
</div>
`;

exports[`should render correctly with given className 1`] = `
<div
  className="month-navigation my-class"
>
  <button
    onClick={[Function]}
  >
    <IconUp />
  </button>
  <button
    onClick={[Function]}
  >
    <IconDown />
  </button>
</div>
`;
